### PR TITLE
Fix support for negative states + Add tests

### PR DESCRIPTION
**Pull request by pedromxavier**
This PR aims to address #157 and supersede #158.

### CODE REVIEW

File: src/OpenBinary/reader.jl [hunk 1]
```diff
@@ -2,10 +2,13 @@ Base.@kwdef mutable struct Reader <: PSRI.AbstractReader
     io::IOStream
 
     # stages
-    stage_total::Int
-    first_year::Int
-    first_stage::Int
     first_relative_stage::Int
```
Comment (raphasampaio) [inline src/OpenBinary/reader.jl:5]: Do we still need the first_relative_stage field?

File: src/OpenBinary/reader.jl [hunk 2]
```diff
@@ -2,10 +2,13 @@ Base.@kwdef mutable struct Reader <: PSRI.AbstractReader
...
+    first_stage::Int
+    last_stage::Int
+    num_stages::Int
+    stage_total::Int
```
Comment (raphasampaio) [inline src/OpenBinary/reader.jl:9]: Do we need both num_stages and stage_total fields?

**Review by raphasampaio — CHANGES_REQUESTED** [review]

File: src/OpenBinary/reader.jl (hunk 1 above)
Comment (pedromxavier) [reply src/OpenBinary/reader.jl:5]: Probably not

File: src/OpenBinary/reader.jl (hunk 2 above)
Comment (pedromxavier) [reply src/OpenBinary/reader.jl:9]: It was important for normalizing the names while keeping track of their original meanings. I think we should discuss them before sticking to a new set of aliases and dropping the unnecessary ones.

**Review by raphasampaio — APPROVED** [review]

**Review by raphasampaio — APPROVED** [review]

**Pull request merged**